### PR TITLE
Added changes for clu helps

### PR DESCRIPTION
**Pull request by arroyc**
added changes needed for finding helps/commands etc for clui

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@arroyc__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (arroyc). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs [hunk 1]
```diff
@@ -183,17 +183,42 @@ public string FormatParameterName(string name, string type, bool isMandatory, bo
             return builder.ToString();
         }
 
-        private void PresentCommandHelp(ICommandBinder commandBinder, string[] arguments, bool prefix)
+        private void PresentCommandCompletion(string[] arguments)
+        {
+            foreach (var cmd in Help.CommandDispatchHelper.CompleteCommands(CLUEnvironment.GetPackagesRootPath(), arguments).OrderBy((cmd) => cmd.Commandline))
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs]: line breaks

File: src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs [hunk 2]
```diff
@@ -51,10 +51,34 @@ public static IEnumerable<CommandInfo> CompleteCommands(string pkgRoot, string[]
...
         }
 
-        public static IEnumerable<CommandInfo> FindMatches(string pkgRoot, string[] args, Func<string, bool> matchFunc)
+        public static HelpInfo FindBestHelp(string pkgRoot, string[] args)
+        {
+            var semiColonSeparatedArgs = String.Join(";", args) + ";";
+
+            Func<string, bool> matcher = (hlp) =>
+            {
+                return hlp.Length <= semiColonSeparatedArgs.Length && CommandDispatchHelper.MatchScore(hlp, semiColonSeparatedArgs) == hlp.Length;
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs]: line breaks

File: src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs [hunk 3]
```diff
@@ -51,10 +51,34 @@ public static IEnumerable<CommandInfo> CompleteCommands(string pkgRoot, string[]
...
+            };
+
+            return FindHelpMatches(pkgRoot, args, matcher).OrderByDescending((hi) => CommandDispatchHelper.MatchScore(hi.Discriminators, semiColonSeparatedArgs)).ThenBy((hi) => hi.Discriminators).FirstOrDefault();
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs]: line breaks

File: src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs [hunk 4]
```diff
@@ -145,12 +176,61 @@ public virtual IEnumerable<CommandInfo> GetCommands()
                     return new CommandInfo[] { };
                 }
             }
+
+            public virtual IEnumerable<HelpInfo> GetHelp()
+            {
+                if (System.IO.Directory.Exists(HelpPath))
+                {
+                    return System.IO.Directory.EnumerateFiles(HelpPath, "*.hlp", System.IO.SearchOption.TopDirectoryOnly).Select((f) =>
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs]: Note: if we had System.IOin a using, this line would fit better

File: src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs [hunk 5]
```diff
@@ -183,17 +183,42 @@ public string FormatParameterName(string name, string type, bool isMandatory, bo
...
+        }
+
+        private void PresentCommandHelp(string[] arguments)
         {
             // BUGBUG - NYI!
             if (arguments.Length == 0)
             {
-                var concatenatedArgs = string.Join(" ", arguments);
-                Console.WriteLine($"Look up help corresponding to the arguments '{concatenatedArgs}'");
+                Console.WriteLine("TODO: Present help for Azure itself...");
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs]: I don't thinkw e should write this to the console: let's put a TODO comment in the code instead

**Comment by markcowl** [comment]
@arroyc A couple of changes.  Also a question:  the cmdlet alias attribute was just changed to use spaces instead of semicolons to separate cmdlet name components - have you tested this manually with the latest code?  Does this simplify any of the logic?

File: src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs (hunk 5 above)
Comment (arroyc) [inline src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs]: removed .. I thought .. for demo we wanted it specifically to be in command prompt

**Pull request merged**